docs/pages/api/automatically-register-agents.mdx

```diff
@@ -0,0 +1,1023 @@
+		DialOpts: []grpc.DialOption{
+			grpc.WithReturnConnectionError(),
+		},
+	})
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("Connected to Teleport")
+
+	d, err := docker.NewClientWithOpts(
+		docker.WithAPIVersionNegotiation(),
+	)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("Connected to the Docker daemon")
+
+	return &tokenDemoApp{
+		teleportClient: t,
+		dockerClient:   d,
+	}
+
+}
+```
+
+`client`, which we alias as `teleport` here, is Teleport's library for setting
+up an API client. Our plugin initializes a Teleport client by calling
+`client.LoadIdentityFile` to obtain a `client.Credentials`. It then uses the
+`client.Credentials` to call `client.New`, which connects to the Teleport Proxy
+Service specified in the `Addrs` field using the provided identity file.
+
+In this example, we are passing the `grpc.WithReturnConnectionError()` function
+call to `client.New`, which instructs the gRPC client to return more detailed
+connection errors.
+
+<Admonition type="warning">
+
+This program does not validate your credentials or Teleport cluster address.
+Make sure that: 
+
+- The identity file you exported earlier does not have an expired TTL
+- The value you supplied to the `Addrs` field in `teleport.Config` includes both
+  the host **and** the web port of your Teleport Proxy Service, e.g.,
+  `mytenant.teleport.sh:443`
+
+</Admonition>
+
+The `newTokenDemoApp` function also initializes a Docker client. It uses version
+negotiation (`docker.WithAPIVersionNegotiation`) to avoid errors due to
+mismatches between your client's API version and the Docker daemon's.
+
+### Add the entrypoint
+
+We are now ready to tie our application together with the entrypoint function,
+`main`. Add the following below the `newTokenDemoApp` function:
+
+```go
+func main() {
+	fmt.Println("Starting the application")
+	app := newTokenDemoApp()
+
+	k := time.NewTicker(updateInterval)
+	defer k.Stop()
+	for {
+		<-k.C
+		if err := app.reconcileApps(); err != nil {
+			panic(err)
+		}
+	}
+}
+```
+
+The `main` function calls `newTokenDemoApp` to initialize our API clients. It
+then calls `time.NewTicker`, which returns a Go **channel** that the entrypoint
+will use to run the reconciliation routine. A channel is a Go primitive that
+manages communication between concurrent routines. 
+
+After we create a ticker, the entrypoint receives from the ticker's channel
+(`k.C`) whenever the `updateInterval` elapses. It blocks until it receives from
+the channel, calling `app.reconcileApps` every interval.
+
+## Step 5/5. Test your client application
+
+Now that you have written your demo application, run it to see how Teleport can
+register and deregister resources in your infrastructure to synchronize with
+your service discovery solution.
+
+Make sure you have pulled the Teleport container image so the application can
+create containers based on it:
+
+```code
+$ docker image pull (=teleport.latest_oss_docker_image=)
```

Comment:
It didn't when I ran the demo app. I think the `docker` CLI bundles some actions together that the API client library separates. `docker run` pulls a container, creates it, and executes it, whereas these are separate Docker API client function calls. I thought it would make sense to pull the container manually, since it's a quick step that you only need to do once.